### DISCUSSION TITLE
IconButton with danger kind

### DISCUSSION THREAD

**Discussion by stanley-regnard** [carbon-design-system/carbon#11832] [Release Feedback]
# Update
Linked to:
Issue: https://github.com/carbon-design-system/carbon/issues/11892;
PR: https://github.com/carbon-design-system/carbon/pull/11893; discussion continued here, working as intended: https://github.com/carbon-design-system/carbon/pull/11893#issuecomment-1203163905

## Context
We are currently migrating from v10 to v11, using `@carbon/react 1.7.0`

## Problem
The following code:
```
...
<Button
    hasIconOnly
    className={canDelete === false ? "hidden" : ""}
    disabled={canDelete === false}
    iconDescription={txt.DELETE}
    kind="danger--ghost"
    onClick={() => onDelete(index)}
    renderIcon={DeleteIcon}
    size="sm"
/>
...
```
Outputs the following warning in the console:
```
Warning: Failed prop type: Invalid prop `kind` of value `danger--ghost` supplied to `ForwardRef(IconButton2)`,
expected one of ["primary","secondary","ghost","tertiary"].
```
Displays and functions as intended otherwise. 

## Possible solution
Found this in IconButton/index.js: 
`  kind: PropTypes.oneOf(['primary', 'secondary', 'ghost', 'tertiary']),`

In Button.js:
`  kind: PropTypes.oneOf(ButtonKinds),`

In prop-types/types.js:
`var ButtonKinds = ['primary', 'secondary', 'danger', 'ghost', 'danger--primary', 'danger--ghost', 'danger--tertiary', 'tertiary'];`

I suppose it should be `  kind: PropTypes.oneOf(ButtonKinds),` in IconButton too?

**Comment by tay1orjones** [comment]
> working as intended: https://github.com/carbon-design-system/carbon/pull/11893#issuecomment-1203163905
